src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -762,39 +776,87 @@ const WorkflowView = _.flow(
+                h(Link, { href: this.getSupportLink('360056384631'), ...Utils.newTabLinkProps },
+                  [clickToLearnMore])
+              ]),
+              span({ style: styles.checkBoxSpanMargins }, [
+                h(LabeledCheckbox, {
+                  checked: retryWithMoreMemory,
+                  onChange: v => this.setState({ retryWithMoreMemory: v }),
+                  style: styles.checkBoxLeftMargin
+                }, [' Retry with more memory'])
+              ])
             ]),
-            h(InfoBox, [
-              'Use a reference disk image if available rather than localizing reference inputs. ',
-              h(Link, {
-                href: 'https://support.terra.bio/hc/en-us/articles/360056384631',
-                ...Utils.newTabLinkProps
-              }, ['Click here to learn more.'])
+            div({ style: { display: 'inline-block' } }, [
+              retryWithMoreMemory && span({ style: { margin: '0 0.5rem 0 0.5rem' } }, [
+                h(IdContainer, [
+                  id => h(Fragment, [
+                    label({
+                      htmlFor: id,
+                      style: { ...styles.label, verticalAlign: 'middle' }
+                    }, ['Memory retry factor:']),
+                    div({ style: { display: 'inline-block', marginLeft: '0.25rem' } }, [
+                      h(NumberInput, {
+                        id,
+                        min: 1.1,
+                        max: 10,
+                        step: 0.1,
+                        isClearable: false,
+                        onlyInteger: false,
+                        value: retryMemoryFactor,
+                        style: { width: '5rem' },
+                        onChange: v => this.setState({ retryMemoryFactor: v })
+                      })
+                    ])
+                  ])
+                ])
+              ]),
+              // We show either an info message or a warning, based on whether increasing memory on retries is
+              // enabled and the value of the retry multiplier.
+              (retryWithMoreMemory && retryMemoryFactor > 2 ?
+                h(InfoBox,
+                  { style: { color: colors.warning() }, iconOverride: 'warning-standard' }, [
+                    'Retry factors above 2 are not recommended. The retry factor compounds and may substantially increase costs. ',
+                    h(Link, { href: this.getSupportLink('4403215299355'), ...Utils.newTabLinkProps },
+                      [clickToLearnMore])
+                  ]) :
+                h(InfoBox, [
+                  'If a task has a maxRetries value greater than zero and fails because it ran out of memory, retry it with more memory. ',
+                  h(Link, { href: this.getSupportLink('4403215299355'), ...Utils.newTabLinkProps },
+                    [clickToLearnMore])
+                ])
```

Comment:
If you use the above suggestion this block can be moved to the above flex div. You should be able to remove the inline-block styling as well.